reps/2022-10-11-serve-java-http-ingress.md

```diff
@@ -0,0 +1,167 @@
+## Summary
+### General Motivation
+We want to call various methods of java deployment through http proxy. Like the fastapi ingress in python serve.
+### Should this change be within `ray` or outside?
+main `ray` project. Changes are made to Ray Serve module.
+
+## Stewardship
+### Required Reviewers
+@simon-mo
+### Shepherd of the Proposal (should be a senior committer)
+@simon-mo
+
+## Design and Architecture
+### user case
+1. define the model with JAX-RS
+```java
+@Path("user")
+public class UserRestService {
+    @GET
+    @Path("helloWorld")
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    public String helloWorld(String name) {
+        return "hello world, " + name;
+    }
+
+    @GET
+    @Path("paramPathTest/{name}")
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    public String paramPathTest(@PathParam("name")String name) {
+        return "paramPathTest, " + name;
+    }
+}
+```
+2. create deployment
+```java
+Deployment deployment =
+        Serve.deployment()
+            .setName("deploymentName")
+            .setDeploymentDef(UserRestService.class.getName())
+            .ingress("jax-rs")
+            .setNumReplicas(1)
+            .create();
+deployment.deploy(true);
+```
+3. calling deployments via HTTP
+The URI is determined by the deployment name and JAX-RS @PATH annotations.
+```java
+curl http://127.0.0.1:8000/deploymentName/user/helloWorld?name=test
+curl http://127.0.0.1:8000/deploymentName/user/paramPathTest/test
+```
+### http ingress
+#### ingress api annotation: JAX-RS
+In java application development, restful api is generally implemented through two sets of annotations, spring web or jax-rs.
```

Comment:
Add link of spring web and jax-rs official webpage.

Reply:
Resolved.
For more information, please click this link: https://projects.eclipse.org/projects/ee4j.rest